### PR TITLE
Add missing attributes to ipasudorule.

### PR DESCRIPTION
**Pull request by rjeffman**
This patch adds the following attributes to ipasudorule:

    - order
    - sudooption
    - runasuser
    - runasgroup

### CODE REVIEW

**Comment by rjeffman** [comment]
This patch fixes issue #158.

File: plugins/modules/ipasudorule.py [hunk 1]
```diff
@@ -91,6 +91,23 @@
     description: Cammand category the sudo rule applies to
```
Comment (t-woerner) [inline plugins/modules/ipasudorule.py]: Please fix the typo in Cammand

File: plugins/modules/ipasudorule.py [hunk 2]
```diff
@@ -212,13 +243,21 @@ def main():
             hostgroup=dict(required=False, type='list', default=None),
             user=dict(required=False, type='list', default=None),
             group=dict(required=False, type='list', default=None),
-            cmd=dict(required=False, type="list", default=None),
+            allow_sudocmdgroups=dict(required=False, type="list", default=None),
```
Comment (t-woerner) [inline plugins/modules/ipasudorule.py]: This should be `allow_sudocmd`

File: plugins/modules/ipasudorule.py [hunk 3]
```diff
@@ -212,13 +243,21 @@ def main():
...
+            deny_sudocmds=dict(required=False, type="list", default=None),
```
Comment (t-woerner) [inline plugins/modules/ipasudorule.py]: Why is this plural?

File: plugins/modules/ipasudorule.py [hunk 4]
```diff
@@ -256,8 +295,16 @@ def main():
     hostgroup = module_params_get(ansible_module, "hostgroup")
     user = module_params_get(ansible_module, "user")
     group = module_params_get(ansible_module, "group")
-    cmd = module_params_get(ansible_module, 'cmd')
-    cmdgroup = module_params_get(ansible_module, 'cmdgroup')
+    allow_sudocmds = module_params_get(ansible_module, 'allow_sudocmds')
```
Comment (t-woerner) [inline plugins/modules/ipasudorule.py]: *_sudocmds and *_sudocomdgroups should all be singular. Also later on.

File: plugins/modules/ipasudorule.py (hunk 1 above)
Comment (t-woerner) [inline plugins/modules/ipasudorule.py]: Also the `cmd` and `cmdgroup` options need to removed and the new options `allow_sudocmd`, `allow_sudocmdgroup`,  `deny_sudocmd` and `deny_sudocmdgroup` need to be added here.

File: tests/sudorule/test_sudorule.yml [hunk 5]
```diff
@@ -190,25 +237,77 @@
     register: result
     failed_when: result.changed
 
-  - name: Ensure sudorule sudocmds are absent
+  - name: Ensure sudorule is present, with an allow_sudocmdgroup.
```
Comment (t-woerner) [inline tests/sudorule/test_sudorule.yml]: Here and in the following uses it seems to be singular already.

File: plugins/modules/ipasudorule.py (hunk 2 above)
Comment (rjeffman) [reply plugins/modules/ipasudorule.py]: Fixed.

File: plugins/modules/ipasudorule.py (hunk 3 above)
Comment (rjeffman) [reply plugins/modules/ipasudorule.py]: Fixed.

File: plugins/modules/ipasudorule.py (hunk 4 above)
Comment (rjeffman) [reply plugins/modules/ipasudorule.py]: Fixed

File: plugins/modules/ipasudorule.py (hunk 1 above)
Comment (rjeffman) [reply plugins/modules/ipasudorule.py]: Fixed.

Comment (rjeffman) [reply plugins/modules/ipasudorule.py]: done.

**Comment by sigbjornaib** [comment]
I am looking for the features provided by this PR, so I went ahead and tested it. 

When using the "allow_sudocmd" attribute, the same sudocmd ends up in both "Sudo Allow Commands" and "Sudo Deny Commands" in the sudorule created on the IPA server.

**Comment by sigbjornaib** [comment]
Doing the actions in a two different tasks seem to work.

First, create the rule in a task with "action: sudorule" and adding the attributes user, host and runasuser. This works as expected.
Second, update the rule in a task with "action: member" and supply only a "allow_sudocmd" attribute.
Now this also works as expected.

However adding "allow_sudocmd" during the first "action: sudorule" task ends up with the same command added to both "Sudo Allow Commands" and "Sudo Deny Commands" in the sudorule created on the IPA server.

**Comment by rjeffman** [comment]
@sigbjornaib, thank you for reporting this.

I fixed the code so that `allow_sudocmd` does work properly with `action: sudorule'.

File: plugins/modules/ipasudorule.py [hunk 6]
```diff
@@ -88,9 +88,26 @@
     required: false
     type: list
   cmdcategory:
```
Comment (t-woerner) [inline plugins/modules/ipasudorule.py]: DOCUMENTATION still contains options cmd and cmdgroup. These should not be there anymore. EXAMPLES also need to be adapted.

File: tests/sudorule/test_sudorule.yml [hunk 7]
```diff
@@ -277,3 +376,26 @@
...
+    ipasudocmdgroup:
+      ipaadmin_password: pass1234
+      name: test_sudorule
+      state: absent
+
+  - name: Ensure hostgroup is absent.
+    ipahostgroup:
+      ipaadmin_password: MyPassword123
+      name: cluster
+      host:
```
Comment (t-woerner) [inline tests/sudorule/test_sudorule.yml]: The use of host with state:absent without action:member will result in an error. Please remove the host.

**Pull request merged**